pkg/kvstore/consul.go

```diff
@@ -87,23 +92,30 @@ func (c *consulModule) getConfig() map[string]string {
 
 func (c *consulModule) newClient() (BackendOperations, error) {
 	if c.config == nil {
-		consulAddr, ok := c.opts[optAddress]
-		if !ok {
+		consulAddr, _ := c.opts[optAddress]
+		configPathOpt, _ := c.opts[consulOptionConfig]
```

Comment:
I'm afraid both of these dictionary lookups need an `ok` check or a nil check. The map values are pointers and we access them below without checking :/

Reply:
@raybejjani opts are always initialized. So the lookup will return true all the time. But I will re-add check condition which make sense. 
https://github.com/cilium/cilium/blob/0fef48623abc1fa36f614dcda2b6629b626d0d14/pkg/kvstore/consul.go#L52

Reply:
you are most correct! sorry! 👍